dist/images/ovnkube.sh

```diff
@@ -475,97 +515,154 @@ ovn-northd () {
 
 # v2 v3 - run ovnkube --master
 ovn-master () {
+  trap 'kill $(jobs -p); exit 0' TERM
   check_ovn_daemonset_version "2 3"
-  # this is only run on masters
-  # ovn-master - master node only
-  echo "=============== ovn-master (wait for ovs) ========== MASTER ONLY"
-  wait_for_ovs
+  rm -f /var/run/openvswitch/ovnkube-master.pid
 
-  echo "=============== ovn-master (wait for northd) ========== MASTER ONLY"
-  wait_for_northd
+  echo "=============== ovn-master (wait for northd_ready) ========== MASTER ONLY"
+  get_master_ovn_vars
+  echo "ovn_nbdb ${ovn_nbdb}   ovn_sbdb ${ovn_sbdb}"
+
+  # wait for northd to start
+  wait_for_event pid_ready ovn-northd.pid
+  sleep 5
 
   echo "=============== ovn-master ========== MASTER ONLY"
+  # use unix socket
+  ovn_nbdb_i=""
+  ovn_sbdb_i=""
   /usr/bin/ovnkube \
-    --init-master ${ovn_host} --net-controller \
-    --cluster-subnet ${OVN_NET_CIDR} --service-cluster-ip-range=${OVN_SVC_CIDR} \
-    --k8s-token=${k8s_token} --k8s-apiserver=${K8S_APISERVER} --k8s-cacert=${K8S_CACERT} \
-    --nb-address=${OVN_NORTH} --sb-address=${OVN_SOUTH} \
+    --init-master ${ovn_pod_host} --net-controller \
+    --cluster-subnet ${net_cidr} --service-cluster-ip-range=${svc_cidr} \
+    --k8s-token=${k8s_token} --k8s-apiserver=${k8s_apiserver} --k8s-cacert=${K8S_CACERT} \
+    --nb-address=${ovn_nbdb_i} --sb-address=${ovn_sbdb_i} \
     --nodeport \
     --loglevel=${ovnkube_loglevel} \
     --pidfile /var/run/openvswitch/ovnkube-master.pid \
     --logfile /var/log/openvswitch/ovnkube-master.log &
   echo "=============== ovn-master ========== running"
+  # time to set up northd and databases
+  wait_for_event pid_ready ovnkube-master.pid
+  sleep 4
+
+  # Listen on 664[12] so ovn-controller and node can start
+  echo "set-connection ptcp:664"
+  ovn-sbctl set-connection ptcp:6642
+  ovn-nbctl set-connection ptcp:6641
```

Comment:
ovnkube --init-master will do these set-connection calls for us. Do we really need them in the script?

Reply:
@dcbw Didn't know where they got set up. I will remove the lines. I wanted to delay setting them up until after master completed initialization.

Reply:
@dcbw in experimenting, 6641 and 6642 are not set up by --init-master, restored lines 550-1

Reply:
@dcbw @pecameron this explicit calling of `set-connection` is required since we are using UNIX scheme to connect to OVN DBs from ovnkube-master.